blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -67,28 +67,30 @@ public static Socket connect(String host, int port, String username, String pass
     }
 
     public static void send(Message message, OutputStream outputStream) throws IOException {
-        log.trace("Writing on: " + outputStream);
-        StringBuffer toSend = new StringBuffer(message.getCommand().toString());
-        toSend.append("\n");
-
-        Iterator<String> keys = message.getHeaders().keySet().iterator();
-        while (keys.hasNext()) {
-            String key = keys.next();
-            String value = message.getHeaders().get(key);
-            toSend.append(key);
-            toSend.append(":");
-            toSend.append(value);
+        synchronized (outputStream) {
```

Comment:
Thanks to point it out and I do agree that it just needs to synchronize on the writes to the socket. I think the write on java.net.Socket is not thread-safe so even if we assemble the data and do a single write, it could not guarantee an atomic operation. The NIO SocketChannel is great and I think it needs to change more codes. So it would be better to do in the other JIRA.